spring-boot/admin/src/main/resources/templates/admin/fragments/api-keys/edit.html

```diff
@@ -227,40 +236,37 @@ <h3> Select API user from table
                 // Add behaviour when the user saves the selection
                 $('#saveApiKeyButton').click(function () {
                     // Check if any github user selection
-                    if (table.row('.selected') !== null && table.row('.selected').data() !== null && typeof table.row('.selected').data() !== 'undefined')
-                    {
+                    if (table.row('.selected') !== null && table.row('.selected').data() !== null && typeof table.row('.selected').data() !== 'undefined') {
                         // Check if any service selection
-                        if ($("#selectedServiceId").val() !== null && $("#selectedServiceId").val() !== "" && typeof $("#selectedServiceId").val() !== 'undefined')
-                        {
-                            // Retrieve the github username which is common attribute in github user database and coatrack user database
-                            $("#selectedUserId").val(table.row('.selected').data().login);
+                        if ($("#selectedServiceId").val() !== null && $("#selectedServiceId").val() !== "" && typeof $("#selectedServiceId").val() !== 'undefined') {
+                            // Retrieve the github/keyrock username which is common attribute in github user database and coatrack user database
+                            ([[${@environment.getProperty('ygg.admin.idm.provider')}]].localeCompare("github") === 0) ? $("#selectedUserId").val(table.row('.selected').data().login) :
+                                $("#selectedUserId").val(table.row('.selected').data().username);
                             $("#apiKeyForm").submit();
-
-                        } else
-                        {
+                        } else {
                             $("#message").html("You must select a service");
                             $("#infoDialog").dialog("open");
                         }
-                    } else
-                    {
+                    } else {
                         $("#message").html("You must select a coatrack user");
                         $("#infoDialog").dialog("open");
                     }
                 });
             });
 
-            function searchGituhbUsers()
-            {
+            function searchGituhbUsers() {
```

Comment:
This should have a generic name, as it is not only for Github anymore.

Reply:
Done, so now it doesn't reference any Github anymore